### ISSUE TITLE
[BUG] Metrics with DataParallel or DistributedDataParallel are not matching the one with models trained on Single GPU

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/Transformers4Rec#495]
### Bug description
When using `Trainer` to train Transformer models, the accuracy obtained with `DataParallel` and `DistributedDataParallel` are much lower to the ones obtained when training the model using single GPU.

Can you pls add the details here @nzarif ?

### Steps/Code to reproduce bug

1. 
2. 
3. 

### Environment details
- Transformers4Rec version: 
- Platform:
- Python version:
- Huggingface Transformers version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):

**Comment by nzarif** [comment]
Training loss is not really affected by using `DataParallel` or `DistributedDataParallel` mode but other metrics such as NDCG and recall will be affected. To resolve the problem, `learning_rate` must be tuned based on the model and the `batch_size`. So in order to get the same metrics as single GPU when using `DataParallel` or `DistributedDataParallel`, `learning_rate` needs to be tuned.
